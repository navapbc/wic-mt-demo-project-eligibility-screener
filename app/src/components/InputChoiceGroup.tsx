import React from 'react'
import styled from 'styled-components'

interface Choice {
  checked: boolean
  handleChange: (e: React.ChangeEvent<HTMLInputElement>) => void
  label: string
  name?: string
  value: string
}

type Props = {
  choices: Choice[]
  title: string
  type: 'checkbox' | 'radio'
}

const InputChoiceGroup = (props: Props): JSX.Element => {
  const { choices, title, type } = props

  return (
    <Fieldset className="usa-fieldset">
      <h2>{title}</h2>
      {choices.map((choice: Choice) => (
        <div className={`usa-${type}`} key={choice.value}>
          <input
            className={`usa-${type}__input`}
            id={choice.value}
            name={choice.name}
            type={type}
            role={type}
            value={choice.value}
            checked={choice.checked}
            onChange={choice.handleChange}
          />
          <label className={`usa-${type}__label`} htmlFor={choice.value}>
            {choice.label}
          </label>
        </div>
      ))}
    </Fieldset>
  )
}

const Fieldset = styled.fieldset`
  border: none;
  h2 {
    font-family: 'Balsamiq Sans', cursive;
    font-weight: 300;
  }
  label {
    font-family: 'Balsamiq Sans', cursive;
    font-weight: 300;
  }
`

export default InputChoiceGroup
