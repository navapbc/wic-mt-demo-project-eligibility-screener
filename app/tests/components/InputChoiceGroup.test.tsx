import { render, screen } from '@testing-library/react'
import { ChangeEvent } from 'react'

import InputChoiceGroup, {
  InputChoiceGroupProps,
} from '@components/InputChoiceGroup'

const choices = [
  {
    checked: false,
    handleChange: (e: ChangeEvent<HTMLInputElement>) => {},
    labelKey: 'label a',
    name: 'name-a',
    value: 'a',
  },
  {
    checked: false,
    handleChange: (e: ChangeEvent<HTMLInputElement>) => {},
    labelKey: 'label b',
    name: 'name-b',
    value: 'b',
  },
  {
    checked: false,
    handleChange: (e: ChangeEvent<HTMLInputElement>) => {},
    labelKey: 'label c',
    name: 'name-c',
    value: 'c',
  },
]

const testProps: InputChoiceGroupProps = {
  choices: choices,
  titleKey: 'title',
  required: false,
  type: 'checkbox',
}

it('should match snapshot when it is a set of checkboxes', () => {
  const { container } = render(<InputChoiceGroup {...testProps} />)
  expect(container).toMatchSnapshot()
})

it('should match snapshot when it is a set of radio buttons', () => {
  const { container } = render(<InputChoiceGroup {...testProps} type="radio" />)
  expect(container).toMatchSnapshot()
})

it('should match display required marker if required is true', () => {
  render(<InputChoiceGroup {...testProps} required={true} />)
  const required = screen.getByText('*')
  expect(required).toBeInTheDocument()
})

it('should display an accordion if passed as props', () => {
  const accordion = {
    headerKey: 'accordion header',
    bodyKey: 'accordion body',
  }
  render(<InputChoiceGroup {...testProps} accordion={accordion} />)
  const accordionElement = screen.getByText('accordion header')
  expect(accordionElement).toBeInTheDocument()
})
