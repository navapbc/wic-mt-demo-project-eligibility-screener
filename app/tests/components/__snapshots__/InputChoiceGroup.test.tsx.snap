// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot when it is a set of checkboxes 1`] = `
<div>
  <fieldset
    class="usa-fieldset"
  >
    <h2>
      title
    </h2>
    <div
      class="usa-checkbox"
    >
      <input
        class="usa-checkbox__input usa-checkbox__input--tile"
        id="name-a-a"
        name="name-a"
        type="checkbox"
        value="a"
      />
      <label
        class="usa-checkbox__label"
        for="name-a-a"
      >
        label a
      </label>
    </div>
    <div
      class="usa-checkbox"
    >
      <input
        class="usa-checkbox__input usa-checkbox__input--tile"
        id="name-b-b"
        name="name-b"
        type="checkbox"
        value="b"
      />
      <label
        class="usa-checkbox__label"
        for="name-b-b"
      >
        label b
      </label>
    </div>
    <div
      class="usa-checkbox"
    >
      <input
        class="usa-checkbox__input usa-checkbox__input--tile"
        id="name-c-c"
        name="name-c"
        type="checkbox"
        value="c"
      />
      <label
        class="usa-checkbox__label"
        for="name-c-c"
      >
        label c
      </label>
    </div>
  </fieldset>
</div>
`;

exports[`should match snapshot when it is a set of radio buttons 1`] = `
<div>
  <fieldset
    class="usa-fieldset"
  >
    <h2>
      title
    </h2>
    <div
      class="usa-radio"
    >
      <input
        class="usa-radio__input usa-radio__input--tile"
        id="name-a-a"
        name="name-a"
        type="radio"
        value="a"
      />
      <label
        class="usa-radio__label"
        for="name-a-a"
      >
        label a
      </label>
    </div>
    <div
      class="usa-radio"
    >
      <input
        class="usa-radio__input usa-radio__input--tile"
        id="name-b-b"
        name="name-b"
        type="radio"
        value="b"
      />
      <label
        class="usa-radio__label"
        for="name-b-b"
      >
        label b
      </label>
    </div>
    <div
      class="usa-radio"
    >
      <input
        class="usa-radio__input usa-radio__input--tile"
        id="name-c-c"
        name="name-c"
        type="radio"
        value="c"
      />
      <label
        class="usa-radio__label"
        for="name-c-c"
      >
        label c
      </label>
    </div>
  </fieldset>
</div>
`;
